modules/rostests/apitests/ntdll/NtQueryInformationProcess.c

```diff
@@ -323,6 +325,103 @@ Test_ProcessWx86Information(void)
     trace("VdmPower = %lu\n", VdmPower);
 }
 
+static
+void
+Test_ProcQueryAlignmentProbe(void)
+{
+    NTSTATUS Status;
+    ULONG InfoClass;
+
+    /* Iterate over the process info classes and begin the tests */
+    for (InfoClass = 0; InfoClass < _countof(PsProcessInfoClass); InfoClass++)
+    {
+        /* The buffer is misaligned */
+        Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                           InfoClass,
+                                           (PVOID)(ULONG_PTR)(PsProcessInfoClass[InfoClass].AlignmentQUERY / 2),
```

Comment:
Note that this trick won't work if AlignmentQUERY is 1. If a case like that exists, you may want to skip over this call instead (since we should never get STATUS_DATATYPE_MISALIGNMENT in that case)